app/models/jobs-model.js

```diff
@@ -9,16 +10,25 @@ const sortRef = {
   employer: { field: 'employer', direction: 'ASC' },
 };
 
+const historicDate = {
+  week: moment().subtract(moment.duration(1, 'w')),
+  fortnight: moment().subtract(moment.duration(2, 'w')),
+  month: moment().subtract(moment.duration(1, 'M')),
+  none: moment('2016-01-01'),
+};
+
 module.exports = db.Model.extend(
   {
     tableName: 'jobs',
     hasTimestamps: true,
   },
   {
-    findAllByAccountId(accountId, sort) {
+    findAllByAccountId(accountId, sort, filter) {
       const sortOrDefault = (sort == null || sort === '') ? 'created' : sort;
+      const filterOrDefault = (filter == null || filter === '') ? 'none' : filter;
```

Comment:
there is no need to check `null and ''` http://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in